cups/request.c

```diff
@@ -1196,8 +1197,8 @@ _cupsSetHTTPError(http_status_t status)	/* I - HTTP status code */
         break;
 
     case HTTP_STATUS_ERROR :
-	_cupsSetError(IPP_STATUS_ERROR_INTERNAL, strerror(errno), 0);
-        break;
+	_cupsSetError(IPP_STATUS_ERROR_INTERNAL, http && http->error != 0 ? strerror(http->error) : "Internal Server Error", 0);
```

Comment:
and here assume that `http` is not `NULL`, which we can do because a) it is a private API and b) all of the callers already ensure the pointer is not `NULL`.

Reply:
Ok, I was thinking around the lines what static analyzers can complaint about and protecting the code even from ourselves (I can forget that I should not pass NULL there, so a protection against it :D ), but yeah, we can assume that I don't forget :D .